pyacq/core/tools.py

```diff
@@ -52,13 +52,22 @@ def process_data(self, pos, data):
         self.new_data.emit(self._pos, data)
     
     def stop(self):
-        with self.lock:
+        with self.running_lock:
             self.running = False
     
     def pos(self):
         return self._pos
 
-
+class ThreadPollOuput(ThreadPollInput):
```

Comment:
I think the name is ok. I just meant that "Output" is misspelled.

How about this for a docstring:

```
Thread that monitors an OutputStream in the background and emits a Qt signal
when data is sent.

Like ThreadPollInput, this class can be used where low-latency response to data
is needed within a Qt main thread (because polling from the main thread with
QTimer either introduces too much latency or consumes too much CPU).
    
The `process_data()` method may be reimplemented to define other behaviors.
"""

```